digest/utility.py

```diff
@@ -117,16 +117,21 @@ def get_missing_required_columns(bagel: pd.DataFrame, schema_file: str) -> set:
     )
 
 
-def get_event_id_columns(bagel: pd.DataFrame, schema: str) -> Union[list, str]:
-    """Returns names of columns which identify a unique assessment or processing pipeline."""
+def get_event_id_columns(bagel: pd.DataFrame, schema: str) -> Union[str, list]:
+    """
+    Returns name(s) of columns which identify a unique assessment or processing pipeline.
+
+    When there is only one relevant column, we return a string instead of a list to avoid grouper problems when the column name is used in pandas groupby.
+    """
     if schema == "imaging":
         return ["pipeline_name", "pipeline_version"]
-    elif schema == "phenotypic":
+    if schema == "phenotypic":
```

Comment:
this has no impact on the function, correct? i.e. The extended docstring is unrelated

Reply:
yes, that's right